src/LoginServer/Network/LoginPacketHandler.cs

```diff
@@ -429,5 +436,23 @@ public void CB_REQ_CHANNEL_TRAFFIC(LoginConnection conn, Packet packet)
 		{
 			Send.BC_NORMAL_ZoneTraffic(conn);
 		}
+
+		/// <summary>
+		/// Sent when the user clicks the barrack number.
+		/// </summary>
+		/// <param name="conn"></param>
+		/// <param name="packet"></param>
+		[PacketHandler(Op.CB_SELECT_BARRACK_LAYER)]
+		public void CB_SELECT_BARRACK_LAYER(LoginConnection conn, Packet packet)
+		{
+			var layer = packet.GetInt();
+
+			// Only these two layers are valid currently.
+			if (layer < 1 || layer > 2)
+				return;
```

Comment:
I love how you can simply return in ToS. In Mabi almost every packet has a response packet, and if you don't send it, the client locks up until it receives it... Though there should be a log message, so that if something about this ever changes, or there is some reason this might have another value somehow, there will be a warning in the console and the log files, which we can look up to figure out why a user is reporting that barrack layer selection isn't working^^ Log all the things, better safe than sorry.

Reply:
Yes, there should be a log message.